src/server/GameController.cpp

```diff
@@ -67,7 +66,11 @@ void GameController::updateState() {
     }
 
     // Detect and handle group collisions
-    mCollisionDetector->detectAndHandleShapeCollisions(mGroups);
```

Comment:
this part got a little uglier / more inefficient, that's why I suggested the alternative way in my other comment